### ISSUE TITLE
✨ Permettre la conversion de Speaker setup avec des signaux en entrée directe (live) et pas seulement avec des fichiers audio en lecture (différé)

### ISSUE THREAD

**Issue by led78** [GRIS-UdeM/SpatGRIS#406]
On aimerait beaucoup utiliser la fonction de conversion d'un Speaker setup A, en entrée, vers un Speaker setup B, en sortie, rendu possible en mode standalone grâce au Player de SpatGRIS, mais avec des signaux en entrée directe plutôt que des fichiers audio en lecture.
Actuellement, on doit avoir une session DAW avec un ControlGRIS établissant une position fixe pour chaque source en entrée directe.
Dans notre cas, notre système de son comporte plus de 64 canaux, mais la plupart des interfaces audio, comme DVS (Dante) sont limités à 64 canaux de sortie. Or, nous avons des artistes qui utilisent SpatGRIS sur leur système, en direct, mais ne peuvent pas utiliser un le Speaker setup de la Satosphère en sortie avec DVS car il comporte plus de 64 canaux.
En revanche, notre serveur audio mac Pro aura ce qu'il faut pour sortir autant de canaux (interface audio DAD Core 256 canaux). On aimerait donc pouvoir convertir directement un Speaker setup virtuel de 64 sorties (DVS de l'artiste) vers les 93.5 canaux de la Satosphère, via notre serveur audio mac Pro, grâce à cette fonction de conversion de SpaGRIS standalone, mais en passthrough direct (sans enregistrement, ni relecture).

**Comment by led78** [comment]
Après vérification, suite à une suggestion de @NicolaGiannini (autour d'une bière à l'Isle de garde -- c'est là qu'on change le monde), j'ai testé une entrée microphone live directement dans SpatGRIS en mode Player et ça fonctionn!  ... Presque.

Contexte :
- On a un artiste qui vient faire une performance live dans la Satosphère et on a très peu de temps à lui consacrer, puisque ce n'est pas une résidence;
- Il sort 4 canaux (quad) qu'il souhaite répartir en quad, mais sur l'ensemble des haut-parleurs de la Satosphère;
- Cas d'usage idéal pour un serveur de spatialisation permettant la conversion de dispositif de haut-parleurs, en l'occurence : input live en quad (4 canaux) vers dôme (93.5 canaux);

Test : 
1. Ouvrir SpatGRIS seulement (pas de DAW, ni ControlGRIS) et connecter l'interface audio recevant les 4 canaux audio live de l'artiste en entrée de SpatGRIS;
2. Utiliser l'interface audio pour le dôme en sortie de SpatGRIS;
3. Charger le Speaker setup quad de l'artiste dans SpatGRIS et faire un bref enregistrement des sorties;
4. Sauvegarder le Speaker setup et les fichiers audio enregistrés dans un même dossier;
5. Ouvrir le Player et charger le dossier, mais ne pas démarrer la lecture et diminuer la fenêtre du Player;
6. Les signaux entrant dans SpatGRIS devraient être visible et rediffusés aux positions indiquées par le Speaker setup d'enregistrement, mais diffusés sur un seul canal de sortie haut-parleur (1:1) et sans interpolation;
7. Ouvrir un DAW et créer un plugin ControlGRIS avec 4 sources, correspondant aux même 4 canaux live;
8. Fermer la fenêtre Player;
9. Vous pouvez alors jouer avec l'étendu horizontal et vertical des sources.

Propositions : 
1. Permettre d'utiliser la conversion de speaker setup, sans nécissiter le chargement de fichiers audio pré-enregistrés au (sans fenêtre de Player finalement);
2. Permettre d'utiliser le potentiomètre d'interpolation pour ajuster la redistribution générale des signaux en entrée sur plusieurs haut-parleurs, sans avoir à ouvrir un DAW avec une instance de ControlGRIS pour ce faire.

**Comment by led78** [comment]
[IMG_9618.zip](https://github.com/GRIS-UdeM/SpatGRIS/files/11771955/IMG_9618.zip)

**Comment by Acousmachin** [comment]
Oui, pouvoir utiliser SpatGris comme un outil de conversion directe serait certainement très utile.
Et aussi de pouvoir envoyer plus de 8 canaux par instance du plugin ! 
La remarque dans le manuel comme quoi 8 canaux c'est ce que proposent la plupart des DAWs me semble dater de quelques années ;-)

**Comment by led78** [comment]
Il y a une raison pour laquelle on ne dépasse pas 8 sources par plugin, l’écriture des automatisations ne tient compte que de la première source et toutes les autres y sont liés. Autrement, les DAW n’arrivent pas à écrire simultanément les positions de plusieurs sources.

Cela dit, ce serait bien d’avoir un contrôle (ControlGRIS) en standalone avec SpatGRIS (sans automatisation) simplement pour permettre des rotations et des repositionnement groupés d’un dispositif virtuel. Ça éviterait de devoir ouvrir un DAW sans audio, juste pour avoir une instance de ControlGRIS qui contrôle la position des sources en entrée directe dans SpatGRIS (moins de latence que via un DAW).

**Comment by DavidPiazza** [comment]
Je me met là-dessus. J'ai quelques idées pour un ControlGRIS standalone...

**Comment by Normandeau** [comment]
On pourrait imaginer que cette version «standalone» puisse s'intégrer directement dans SpatGRIS par exemple. Une sorte de spatialisateur, ou plutôt de positionneur, statique.

**Comment by led78** [comment]
> Je me met là-dessus. J'ai quelques idées pour un ControlGRIS standalone...

Je peux vous communiquer des cas d’usage très concrets et actuels pour cela. Je suis un peu dans le jus avec Mutek, mais je vous reviens après!

**Comment by led78** [comment]
Précision de la demande : permettre l'ouverture d'une instance ControlGRIS autonome (hors DAW), sans limite au nombre de canaux contrôlé puisqu'aucune écriture d'automatisaion.

Avoir la possibilité d'y importer les positions de chaque source à partir d'un speaker setup, à l'instar du Multi-Encoder de la suite IEM :
![image](https://github.com/GRIS-UdeM/SpatGRIS/assets/13606544/7e6a2f76-55c8-4880-8657-48d184cdca14)
